lib/tasks_private/instance_types.rake

```diff
@@ -190,14 +190,20 @@ EOT
 
   new_instance_types_file << output_instances.map { |i| format_instance(i) }.join("\n")
   new_instance_types_file << <<EOT
-  }
+  }.freeze
+
+  def self.instance_types
```

Comment:
Is this copied from above?  Can we use that method instead of duplicating it?

Reply:
You mean the `def self.instance_types` method in `instance_types.rb`?  
The `instance_types.rb` is the output of this rake task

Reply:
I find it weird in general that this generates source code.  I feel this task should just generate a YAML file, which is processed on first access by the code.  Then this "duplication" wouldn't exist either.

Reply:
I feel strongly the same as well.

My plan is to finish this task first for 4.6 and then revisit this.